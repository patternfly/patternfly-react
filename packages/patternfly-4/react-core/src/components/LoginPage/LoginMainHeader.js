import React from 'react';
import PropTypes from 'prop-types';
import { Title } from '../Title';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/patternfly/components/Login/login.css';

const propTypes = {
  /** Content rendered inside the Login Main Header */
  children: PropTypes.node,
  /** Additional classes added to the Login Main Header */
  className: PropTypes.string,
  /** Title for the Login Main Header */
  title: PropTypes.string,
  /** Subtitle that contains the Text, URL, and URL Text for the Login Main Header */
  subtitle: PropTypes.node,
  /** Additional props are spread to the container <header> */
  '': PropTypes.any
};

const defaultProps = {
  children: null,
  className: '',
  title: '',
  subtitle: ''
};

const LoginMainHeader = ({ children, className, title, subtitle, ...props }) => (
  <header className={css(styles.loginMainHeader, className)} {...props}>
    {title && <Title size="3xl">{title}</Title>}
    {subtitle && <p>{subtitle}</p>}
    {children}
  </header>
);

LoginMainHeader.propTypes = propTypes;
LoginMainHeader.defaultProps = defaultProps;

export default LoginMainHeader;
