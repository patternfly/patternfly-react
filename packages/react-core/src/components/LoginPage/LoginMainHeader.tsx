import * as React from 'react';
import { Title, TitleSizes } from '../Title';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/Login/login';

export interface LoginMainHeaderProps extends React.HTMLProps<HTMLDivElement> {
  /** Content rendered inside the Login Main Header */
  children?: React.ReactNode;
  /** Additional classes added to the Login Main Header */
  className?: string;
  /** Title for the Login Main Header */
  title?: string;
  /** Subtitle that contains the Text, URL, and URL Text for the Login Main Header */
  subtitle?: string;
}

export const LoginMainHeader: React.FunctionComponent<LoginMainHeaderProps> = ({
  children = null,
  className = '',
  title = '',
  subtitle = '',
  ...props
}: LoginMainHeaderProps) => (
  <header className={css(styles.loginMainHeader, className)} {...props}>
    {title && (
      <Title headingLevel="h2" size={TitleSizes['3xl']}>
        {title}
      </Title>
    )}
    {subtitle && <p className={css(styles.loginMainHeaderDesc)}>{subtitle}</p>}
    {children}
  </header>
);
LoginMainHeader.displayName = 'LoginMainHeader';
