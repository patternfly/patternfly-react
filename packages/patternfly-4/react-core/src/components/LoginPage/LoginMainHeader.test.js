import React from 'react';
import LoginMainHeader from './LoginMainHeader';
import { shallow } from 'enzyme';

test('renders with PatternFly Core styles', () => {
  const view = shallow(<LoginMainHeader />);
  expect(view).toMatchSnapshot();
});

test('className is added to the root element', () => {
  const view = shallow(<LoginMainHeader className="extra-class" />);
  expect(view.prop('className')).toMatchSnapshot();
});

test('extra props are spread to the root element', () => {
  const testId = 'login-header';
  const view = shallow(<LoginMainHeader data-testid={testId} />);
  expect(view.prop('data-testid')).toBe(testId);
});
