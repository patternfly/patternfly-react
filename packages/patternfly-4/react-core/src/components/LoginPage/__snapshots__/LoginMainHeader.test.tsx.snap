// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`className is added to the root element 1`] = `"pf-c-login__main-header extra-class"`;

exports[`renders with PatternFly Core styles 1`] = `
<header
  className="pf-c-login__main-header"
/>
`;

exports[`title and subtitle are rendered correctly 1`] = `"pf-c-login__main-header"`;
